### PR TITLE
Add workflow error notification in ui

### PR DESCRIPTION
**Pull request by yp05327**
image
image

### CODE REVIEW

File: templates/repo/actions/list.tmpl [hunk 1]
```diff
@@ -9,7 +9,17 @@
...
-						<a class="item{{if eq .Name $.CurWorkflow}} active{{end}}" href="{{$.Link}}?workflow={{.Name}}">{{.Name}}</a>
+						<a class="item{{if eq .Entry.Name $.CurWorkflow}} active{{end}}" href="{{$.Link}}?workflow={{.Entry.Name}}">{{.Entry.Name}}
+							{{if .IsInvalid}}
+								<span class="tooltip gt-vm" data-content="{{$.locale.Tr "actions.runs.invalid_workflow_helper"}}">
+									<i class="warning icon red"></i>
+								</span>
+							{{else}}
+								<span class="tooltip gt-vm" data-content="{{$.locale.Tr "actions.runs.valid_workflow_helper"}}">
+									<i class="check icon green"></i>
+								</span>
```
Comment (techknowlogick) [inline templates/repo/actions/list.tmpl]:
```suggestion
								</span>
```

**Comment by lunny** [comment]
Could we display the parse error on the tooltip?

File: routers/web/repo/actions/actions.go [hunk 2]
```diff
@@ -62,13 +67,28 @@ func List(ctx *context.Context) {
...
+			workflow := WorkFlow{Entry: *entry}
+			content, err := actions.GetContentFromEntry(entry)
+			if err != nil {
+				ctx.Error(http.StatusInternalServerError, err.Error())
+				return
+			}
+			_, err = actions.GetEventsFromContent(content)
+			if err != nil {
+				workflow.IsInvalid = true
+			} else {
```
Comment (Zettat123) [inline routers/web/repo/actions/actions.go]: This `else` is unnecessary because the default value of `IsInvalid` is `false`.

**Comment by yp05327** [comment]
> Could we display the parse error on the tooltip?

@lunny 
I think we can add `ErrorType` to `WorkFlow ` struct to display the error. 
But I have no idea about how to define these parse errors.

``` go
type WorkFlow struct {
    Entry     git.TreeEntry
    IsInvalid bool
    ErrorType WorkFlowError
}

type WorkFlowError string

const (
    ParseError1 WorkFlowError = "xxxx"
)
```

**Comment by lunny** [comment]
How about
```go
type WorkFlow struct {
Entry     git.TreeEntry
ErrMsg string
}
```

**Comment by yp05327** [comment]
It looks as following now.
image
image

File: routers/web/repo/actions/actions.go [hunk 3]
```diff
@@ -23,6 +23,12 @@ const (
 	tplViewActions base.TplName = "repo/actions/view"
 )
 
+type WorkFlow struct {
```
Comment (wolfogre) [inline routers/web/repo/actions/actions.go]: Why not `Workflow`?

Just a question, not blocking.

**Review by wolfogre — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: routers/web/repo/actions/actions.go (hunk 3 above)
Comment (yp05327) [reply routers/web/repo/actions/actions.go]: You are right. `WorkFlow` is strange.

**Pull request merged**